--- a/scripts/buildorder.py
+++ b/scripts/buildorder.py
@@ -50,8 +50,10 @@
                 for dependency_value in re.split(',|\\|', dependencies_string):
                     # Replace parenthesis to ignore version qualifiers as in "gcc (>= 5.0)":
                     dependency_value = re.sub(r'\(.*?\)', '', dependency_value).strip()
+                    # Handle dependencies on *-dev packages:
+                    dependency_value = re.sub('-dev$', '', dependency_value)
                     arch = os.getenv('TERMUX_ARCH')
                     if arch is None:
                         arch = 'aarch64'
                     if arch == "x86_64":
                         arch = "x86-64"
