--- a/scripts/build/toolchain/termux_setup_toolchain_23c.sh
+++ b/scripts/build/toolchain/termux_setup_toolchain_23c.sh
@@ -109,6 +109,10 @@
 	local env_host="${CARGO_TARGET_NAME//-/_}"
 	export CARGO_TARGET_${env_host@U}_LINKER="${CC}"
 	export CARGO_TARGET_${env_host@U}_RUSTFLAGS="-L${TERMUX__PREFIX__LIB_DIR} -C link-arg=-Wl,-rpath=${TERMUX__PREFIX__LIB_DIR} -C link-arg=-Wl,--enable-new-dtags"
+	if [ "$TERMUX_ARCH" = "aarch64" ]; then
+		# Legacy compatibility: align Rust binaries to 16KB pages
+		export CARGO_TARGET_${env_host@U}_RUSTFLAGS+=" -C link-arg=-Wl,-z,max-page-size=16384"
+	fi
 	export CFLAGS_${env_host}="${CPPFLAGS} ${CFLAGS}"
 	export CC_x86_64_unknown_linux_gnu="gcc"
 	export CFLAGS_x86_64_unknown_linux_gnu="-O2"
--- a/scripts/build/toolchain/termux_setup_toolchain_29.sh
+++ b/scripts/build/toolchain/termux_setup_toolchain_29.sh
@@ -110,6 +110,10 @@
 	local env_host="${CARGO_TARGET_NAME//-/_}"
 	export CARGO_TARGET_${env_host@U}_LINKER="${CC}"
 	export CARGO_TARGET_${env_host@U}_RUSTFLAGS="-L${TERMUX__PREFIX__LIB_DIR} -C link-arg=-Wl,-rpath=${TERMUX__PREFIX__LIB_DIR} -C link-arg=-Wl,--enable-new-dtags"
+	if [ "$TERMUX_ARCH" = "aarch64" ]; then
+		# Legacy compatibility: align Rust binaries to 16KB pages
+		export CARGO_TARGET_${env_host@U}_RUSTFLAGS+=" -C link-arg=-Wl,-z,max-page-size=16384"
+	fi
 	export CFLAGS_${env_host}="${CPPFLAGS} ${CFLAGS}"
 	export CC_x86_64_unknown_linux_gnu="gcc"
 	export CFLAGS_x86_64_unknown_linux_gnu="-O2"
